File: 00._Tasks/03a._Data_parsing_server/node_server/fileReaders/read_csv_file.js
import csv from "csvtojson";
import { Person } from "../models/person.js";
import { getFileURLPath } from "../util.js";

export const formatCSVFileToJson = async (fileName) => {
  const jsonObj = await csv().fromFile(getFileURLPath(fileName));

  const people = jsonObj.map((person) => {
    return new Person(
      `${person.firstname} ${person.lastname}`,
      Number(person.age),
      Number(person.heightInCm),
      person.hobbies.split(",").map(hobby => hobby.trim()),  // Splitting and trimming hobbies into an array
      {
        street: person.address_street,
        city: person.address_city,
        zipcode: person.address_zipcode
      }
    );
  });

  return people;
};

File: 00._Tasks/03a._Data_parsing_server/python_server/read_csv_file.py
import csv
from Person import Person

def read_csv(file_path):
    data = []
    with open(file_path, newline='') as csvfile:
        reader = csv.DictReader(csvfile)
        for row in reader:
            # Creating Person object for each row
            person = Person(
                name=f"{row['firstname']} {row['lastname']}",
                age=int(row['age']),
                height_in_cm=int(row['heightInCm']),
                hobbies=[hobby.strip() for hobby in row['hobbies'].split(',')],
                address={
                    'street': row['address_street'],
                    'city': row['address_city'],
                    'zipcode': row['address_zipcode']
                }
            )
            data.append(person)
    return data
